File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool cmp(string &a, string &b){
        return (a+b)>=(b+a);
    }
    string largestNumber(vector<int>& nums) {
        vector<string>s(nums.size(),"");
        int x=0;
        for(int i=0; i<nums.size(); i++){//keep count for zeros
           if(nums[i]==0){
               x++;
           }
        }
        if(x==nums.size()) return "0";//if all elements are 0 we return "0"
        
        for(int i=0; i<nums.size(); i++){//put all the int to string vector
            s[i]=to_string(nums[i]);
        }
        sort(s.begin(),s.end(),cmp);//sort with custom comparator
        string ans="";
        for(int i=0; i<s.size(); i++){//append the answer to a string
            ans+=s[i];
        }
        return ans;
    }
};

File: 0179-largest-number/0179-largest-number.py
class Solution:
    def customComparator(self, a: str, b: str):
        ans = - 1
        if (a+b) < (b+a):
            ans = 1
        elif (a+b) == (b+a):
            ans = 0
        
        return ans
        
    def largestNumber(self, nums: List[int]) -> str:
        length = len(nums)
        list_string = []
        # count_zero = 0
        for num in nums:
            # if num == 0:
            #     count_zero += 1
            list_string.append(str(num))
            
        # if count_zero == length:
        #     return "0"
        
        sorted_list = sorted(list_string, key = cmp_to_key(self.customComparator));
        # list_string.sort(key = cmp_to_key(self.customComparator));
        
        String =str(int("".join(sorted_list)))
            
        return String
            
        